### ISSUE TITLE
Model parallelism

### ISSUE THREAD

**Issue by llmgpt** [IDEA-Research/DINO#267]
How to use the model parallel way, my picture resolution is very large? In my attempt to split, loss.backward no longer has the same GPU, or optim no longer has the same gpu.
